File: 12 days of Christmas/Haskell/main.hs
main=do mapM(putStrLn)(zipWith(\x a->x++a++"\n")(map(\x->"On the "++x++" day of Christmas\nMy true love sent to me\n")(lines"First\nSecond\nThird\nFourth\nFifth\nSixth\nSeventh\nEighth\nNinth\nTenth\nEleventh\nTwelfth"))(scanl1(\x a->a++"\n"++x)(lines"A Partridge in a Pear Tree.\nTwo Turtle Doves, and\nThree French Hens,\nFour Calling Birds,\nFive Gold Rings,\nSix Geese-a-Laying,\nSeven Swans-a-Swimming,\nEight Maids-a-Milking,\nNine Ladies Dancing,\nTen Lords-a-Leaping,\nEleven Pipers Piping,\nTwelve Drummers Drumming,")))

File: 12 days of Christmas/Fortran/main.f90
character(99)::s(24)=[character(99)::"First","Second","Third","Fourth","Fifth","Sixth","Seventh","Eighth","Ninth","Tenth",&
"Eleventh","Twelfth","\nTwelve Drummers Drumming,","\nEleven Pipers Piping,","\nTen Lords-a-Leaping,",&
"\nNine Ladies Dancing,","\nEight Maids-a-Milking,","\nSeven Swans-a-Swimming,","\nSix Geese-a-Laying,",&
"\nFive Gold Rings,","\nFour Calling Birds,","\nThree French Hens,","\nTwo Turtle Doves, and","\nA Partridge in a Pear Tree.\n"]
do i=1,12
print'(15a)',"On the ",trim(s(i))," day of Christmas\nMy true love sent to me",s(25-i:24)
end do
end